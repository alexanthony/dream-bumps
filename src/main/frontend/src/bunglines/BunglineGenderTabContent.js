import React, {PropTypes} from 'react';
import {Grid, Cell} from 'material-grid/dist';
import { connect } from 'react-redux';

import CrewList from './CrewList';
import CurrentUserCrew from '../usercrews/CurrentUserCrew';

const BunglineGenderTabContent = ({ sex, authenticated }) => (
  <Grid>
    { authenticated &&
      <Cell col={4}>
        <CurrentUserCrew sex={sex} />
      </Cell>
    }
    <Cell col={8} offset={authenticated ? 0 : 2}>
      <CrewList sex={sex} />
    </Cell>
  </Grid>
);

BunglineGenderTabContent.propTypes = {
  sex: PropTypes.string,
  authenticated: PropTypes.bool
};

const mapStateToProps = state => ({
  authenticated: state.auth.authenticated
});

export default connect(mapStateToProps)(BunglineGenderTabContent);
// export default connect(null, { null })(BunglineGenderTabContent);
