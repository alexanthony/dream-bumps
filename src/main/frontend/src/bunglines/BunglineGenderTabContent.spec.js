import React from 'react';
import { shallow } from 'enzyme';
import { BunglineGenderTabContent } from './BunglineGenderTabContent';
import toJson from 'enzyme-to-json';

test('Renders properly when authenticated', () => {
  const component = shallow(
    <BunglineGenderTabContent
      authenticated
      sex="male"
    />
  );
  expect(component).toBeDefined();
  expect(component.find('Cell').length).toEqual(2);
  expect(component.find('Connect(CurrentUserCrew)').length).toEqual(1);
  expect(toJson(component)).toMatchSnapshot();
});

test('Renders properly when not authenticated', () => {
  const component = shallow(
    <BunglineGenderTabContent
      authenticated={false}
      sex="female"
    />
  );
  expect(component).toBeDefined();
  expect(component.find('Cell').length).toEqual(1);
  expect(component.find('Connect(CurrentUserCrew)').length).toEqual(0);
  expect(toJson(component)).toMatchSnapshot();
});
