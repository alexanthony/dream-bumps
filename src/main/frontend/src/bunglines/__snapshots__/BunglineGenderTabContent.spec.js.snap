exports[`test Renders properly when authenticated 1`] = `
<Grid>
  <Cell
    col={4}>
    <Connect(CurrentUserCrew)
      sex="male" />
  </Cell>
  <Cell
    col={8}
    offset={0}>
    <Connect(CrewList)
      sex="male" />
  </Cell>
</Grid>
`;

exports[`test Renders properly when not authenticated 1`] = `
<Grid>
  <Cell
    col={8}
    offset={2}>
    <Connect(CrewList)
      sex="female" />
  </Cell>
</Grid>
`;
